kong/db/strategies/cassandra/init.lua

```diff
@@ -1461,7 +1461,7 @@ do
         local pager = function(size, offset)
           return strategy[method](strategy, primary_key, size, offset)
         end
-        for row, err in iteration.by_row(self, pager) do
+        for row, err in iteration.by_row(self, pager, 1000) do
```

Comment:
Why not `options.page_size`? And having it being specified/defaulted by the DAO (not the strategy). This will allow to eventually make the page_size configurable (humanly or via heuristics) on a use-case basis.

Reply:
(Such an option would of course be a NOP for relevant PostgreSQL queries - non-SELECT DML queries, that is)

Reply:
> There seems to be a lot of places in which we hard-coded the default/max page size to 1000

That is true. The default was picked up for admin api, and then after that a lot of places we have changed it for performance reasons. Originally it should have defaulted to `1000` and `admin api` should have changed it to `100`.

> Why not options.page_size?

The problem is that sometimes user calls, e.g.:
`dao:page` and that results additional queries. It is a bit problematic to pass that `options` table to those sub-queries too. Here we just optimized the cascade deletes. I think it is fine hard-coded.

Reply:
What I also mean that It is also a bit unintuitive to pass `options.page_size` to `delete`. The cascade deletion is a side effect of `delete`, and we just want to make it faster by iterating a bigger result set at once.

Reply:
> The default was picked up for admin api, and then after that a lot of places we have changed it for performance reasons.

I am very much aware, having initially chosen those values :) I was only mentioning it to say that now that the pattern has spread in a lot of hard-coded magic numbers, we should correct that and introduce a variable. Maybe a constant at first, and something we can configure later on, as stated in my above comment. 

It is better hard-coded as a constant, maybe an attribute of the connector (possibly set in the connector's parent class) or DAO. E.g. `options.page_size_subqueries`.

> I think it is fine hard-coded.

No; besides the obvious gains from removing magic numbers in favor of a named constant, it also doesn't close the door to subsequent performance improvements, if the day comes that we want to make it configurable.

Reply:
Merging as-is for now, we can skip the above improvements for a separate PR with its own scope.

Reply:
@thibaultcha yes, i have commits almost ready for that. I will push tomorrow a new pr.